### PR TITLE
Add some callbacks of read_rect/write_rect/copy_rect/map_mem

### PR DESCRIPTION
**Pull request by yangzexia**
CTS case： bufferreadwrite

(cherry picked from commit 73591ab)

### CODE REVIEW

**Comment by yangzexia** [comment]
#36

File: lib/CL/devices/ventus/pocl_ventus.cc [hunk 1]
```diff
@@ -1101,6 +1101,178 @@ void pocl_ventus_write(void *data,
...
+                       const size_t *__restrict__ const region,
+                       size_t const buffer_row_pitch,
+                       size_t const buffer_slice_pitch,
+                       size_t const host_row_pitch,
+                       size_t const host_slice_pitch)
+{
+    cl_mem_flags flags = src_buf->flags;
+    void *__restrict__ device_ptr = src_mem_id->mem_ptr;
+
+    if (flags & CL_MEM_USE_HOST_PTR) {
```
Comment (Jules-Kong) [inline lib/CL/devices/ventus/pocl_ventus.cc:1118]: Does this situation need to be considered individually? When creating the buffer, did we not write from the host to the device buffer when encountering CL_MEM_USE_HOST_PTR?

File: lib/CL/devices/ventus/pocl_ventus.cc [hunk 2]
```diff
@@ -1101,6 +1101,178 @@ void pocl_ventus_write(void *data,
...
+    char const *__restrict const adjusted_device_ptr
+        = (char const *)device_ptr + buffer_origin[2] * buffer_slice_pitch
+        + buffer_origin[1] * buffer_row_pitch + buffer_origin[0];
+    char *__restrict__ const adjusted_host_ptr
+        = (char *)host_ptr + host_origin[2] * host_slice_pitch
+        + host_origin[1] * host_row_pitch + host_origin[0];
+    size_t j, k;
+
+    /* TODO: handle overlaping regions */
+    if ((buffer_row_pitch == host_row_pitch && host_row_pitch == region[0])
```
Comment (Jules-Kong) [inline lib/CL/devices/ventus/pocl_ventus.cc:1131]: What kind of special situation is this?

File: lib/CL/devices/ventus/pocl_ventus.cc [hunk 3]
```diff
@@ -1101,6 +1101,178 @@ void pocl_ventus_write(void *data,
...
+                        const size_t *__restrict__ const region,
+                        size_t const buffer_row_pitch,
+                        size_t const buffer_slice_pitch,
+                        size_t const host_row_pitch,
+                        size_t const host_slice_pitch)
+{
+    vt_device_data_t *d = (vt_device_data_t *)data;
+    cl_mem_flags flags = dst_buf->flags;
+    void *__restrict__ device_ptr = dst_mem_id->mem_ptr;
+    uint64_t dev_addr = 0;
```
Comment (Jules-Kong) [inline lib/CL/devices/ventus/pocl_ventus.cc:1159]: Can the two lines 1159 and 1161 be merged?

**Review by Jules-Kong — APPROVED** [review]
LGTM

**Pull request merged**